core/src/main/java/io/jenkins/pluginhealth/scoring/probes/AbstractGitHubWorkflowProbe.java

```diff
@@ -0,0 +1,114 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.jenkins.pluginhealth.scoring.probes;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+import io.jenkins.pluginhealth.scoring.model.Plugin;
+import io.jenkins.pluginhealth.scoring.model.ProbeResult;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class AbstractGitHubWorkflowProbe extends Probe {
+
+    public static final String KEY = "abstract-github-workflow";
+    private static final Logger LOGGER = LoggerFactory.getLogger(AbstractGitHubWorkflowProbe.class);
+    private static final String WORKFLOWS_DIRECTORY = ".github/workflows";
+
+    @Override
+    protected ProbeResult doApply(Plugin plugin, ProbeContext context) {
+        final Path repository = context.getScmRepository();
+        final Path workflowPath = repository.resolve(WORKFLOWS_DIRECTORY);
+
+        if (!Files.exists(workflowPath)) {
+            return ProbeResult.failure(key(), "Plugin has no GitHub Action configured");
+        }
+
+        try (Stream<Path> files = Files.find(workflowPath, 1,
+            (path, $) -> Files.isRegularFile(path)
+        )) {
+            final ObjectMapper yaml = new ObjectMapper(new YAMLFactory());
+
+            return files
+                .map(file -> {
+                    try {
+                        return yaml.readValue(Files.newInputStream(file), WorkflowDefinition.class);
+                    } catch (IOException e) {
+                        LOGGER.warn("Couldn't not read {} for {} on {}", file, key(), plugin.getName(), e);
+                        return new WorkflowDefinition(Map.of());
+                    }
+                })
```

Comment:
I would suggest to extract this code block to a dedicated function, so you can Javadoc that function and explain properly what this bit does, and you can then use it in a `map` as a one liner to ease the reading.

Reply:
For Jagruti: you can extract with your IDE the content of the `map` method call into a method. This way you will be able to document it. Make sure to keep it `private` as we don't want it (at least for now) to be re-implemented.